src/migration.types.ts

```diff
@@ -60,3 +60,20 @@ export interface LegacyDashboardState {
   };
   writable: boolean;
 }
+
+export type PageMigrationReport = Record<string, { warnings: string[] }>;
+
+export interface DashboardMigrationReport {
+  name: string;
+  errorStack?: string[];
+  pages?: PageMigrationReport;
```

Comment:
Shouldn't there be a page migration report for each page? If I am reading this correctly, there seems to be a single page migration report in here, which is a map of leaf keys (I assume) to errors.

Reply:
In other words, it seems that a dashboard report contains a map of leaf keys to errors.
Whereas I am expecting it to contain a map of page keys to a map of leaf keys to errors.

Reply:
There is a report for each page, it seems that the name for that type is misleading, I will inline it instead.